src/plugins/intel_cpu/src/node.cpp

```diff
@@ -316,6 +315,121 @@ void Node::selectPreferPrimitiveDescriptor(const std::vector<impl_desc_type>& pr
     selectPrimitiveDescriptorByIndex(0);
 }
 
+bool Node::isOneDimShape(const ov::PartialShape& pshape) {
+    int value_1_num = 0;
+    int sz = static_cast<int>(pshape.size());
+    for (auto s : pshape) {
+        if (s.is_static() && s.get_length() == 1) {
+            value_1_num++;
+        }
+    }
+    return value_1_num >= sz - 1;
+}
+
+bool Node::isReorderRequired(ov::intel_cpu::MemoryDescPtr desc1, ov::intel_cpu::MemoryDescPtr desc2) {
+    bool samePrec = desc1->getPrecision() == desc2->getPrecision();
+    bool isOneDimShape1 = isOneDimShape(desc1->getShape().toPartialShape());
+    bool isOneDimShape2 = isOneDimShape(desc2->getShape().toPartialShape());
+    return !(isOneDimShape1 && isOneDimShape2 && samePrec);
+}
+
+void Node::selectPreferPrimitiveDescriptorWithShape(const std::vector<impl_desc_type>& priority, bool ignoreConstInputs) {
+    auto estimateReorderOverhead = [&](const ov::intel_cpu::NodeDesc& supportedPrimitiveDesc, size_t i) {
+        int estimate = 0;
+        auto inputNodesNum = supportedPrimitiveDesc.getConfig().inConfs.size();
+        for (size_t j = 0; j < inputNodesNum; j++) {
+            auto parentEdge = getParentEdgeAt(j);
+            auto parentPtr = parentEdge->getParent();
+
+            // We don't take into account constant edges since reorders on them will be executed on load network
+            // stage
+            if (ignoreConstInputs && j > 0 && parentPtr->isConstant()) {
+                continue;
+            }
+
+            auto parent_spd = parentPtr->getSelectedPrimitiveDescriptor();
+            if (parent_spd != nullptr && !parent_spd->getConfig().outConfs.empty()) {
+                int inNum = parentEdge->getInputNum();
+                if (inNum < 0 || inNum >= static_cast<int>(parent_spd->getConfig().outConfs.size())) {
+                    inNum = 0;
+                }
+                auto curDesc = supportedPrimitiveDesc.getConfig().inConfs[j].getMemDesc();
+                auto parentDesc = parent_spd->getConfig().outConfs[inNum].getMemDesc();
+
+                const bool isCompatible = curDesc->isCompatible(*parentDesc);
+                if (!isCompatible) {
+                    if (!isReorderRequired(parentDesc, curDesc)) {
+                        estimate += 1;
+                    } else {
+                        estimate += ov::shape_size<ov::intel_cpu::VectorDims>(curDesc->getShape().getStaticDims());
```

Comment:
The algo assumes Subgraph works with static shapes only.
We need to add logic that clarifies the shape is not dynamic - overwise we will get exception here.

Reply:
fix potential dynamic case exception issue, just call `getMinDims`

Reply:
MinDims might contain UNDEFINED_DIM. Will the logic be correct in that case?

Reply:
I did a test, and found dynamic shape with UNDEFINED_DIM will return 0. 
```
auto s1 = ov::intel_cpu::Shape(ov::PartialShape({2, -1, 6}));
std::cout << "s1=" << s1.toString() << ",isDynamic=" << s1.isDynamic()
                          << ", getMinDims=" << ov::util::vector_to_string(s1.getMinDims())
                          << ", getMaxDims=" << ov::util::vector_to_string(s1.getMaxDims()) << std::endl;
```
Result:
`s1={2, 0 - ?, 6},isDynamic=1, getMinDims=[ 2, 0, 6 ], getMaxDims=[ 2, 18446744073709551615, 6 ]`

So I have to adopt old logical if current node has any dynamic input. @dmitry-gorokhov